normalizer/normalizer.bal

```diff
@@ -0,0 +1,64 @@
+import ballerina/http;
+
+type ZendeskResponse record {
+    record {|
+        string url;
+        int id;
+        string subject;
+    |} ticket;
+};
+
+service /api/v1 on new http:Listener(8080) {
+    
+    resource function post ticket(@http:Payload json|xml request) returns xml|json|error {
+        json zendeskRequest = {};
+        if request is json && validateJsonRequest(request) {
+            zendeskRequest = {
+                ticket: {
+                    subject: check request.subject, 
+                    comment: {
+                        body: check request.comment
+                        }
+                    }
+                };
+        }
+        if request is xml && validateXmlRequest(request) {
+            zendeskRequest = {
+                ticket: {
+                    subject: (request/<subject>).data(), 
+                    comment: {
+                        body: (request/<comment>).data()
+                        }
+                    }
+                };
+        }
+        if zendeskRequest != {} {
+            http:Client zendeskClient = check new ("http://api.zendesk.com.balmock.io");
+            ZendeskResponse result = check zendeskClient->/api/v2/tickets.post(zendeskRequest);
+            if request is xml {
+                return xml `
+                    <response>
+                    <id>${result.ticket.id}</id>
+                    <subject>${result.ticket.subject}</subject>
+                    <url>${result.ticket.url}</url>
+                    </response>`;
+            } else {
+                return {
+                    id: result.ticket.id,
+                    subject: result.ticket.subject,
+                    url: result.ticket.url
+                };
+            }
+        }
+        return error ("Invalid request");
+    }       
+}
+
+function validateXmlRequest(xml request) returns boolean =>
+    (request/<subject>).data() == "Printer offline"
+        && (request/<comment>).data() == "My printer is offline. Restarting doesn’t help.";
+
+function validateJsonRequest(json request) returns boolean => {
+        "subject": "Printer offline",
+        "comment": "My printer is offline. Restarting doesn’t help."
+    } == request;
```

Comment:
I don't think we should be validating for the hard coded response here

Reply:
+1, we can't validated against hard-coded value.